### PR TITLE
Add `starts-with`, `ends-with` to `string`

### LINKED ISSUE #9877: Partial sequence comparison for strings
### Feature Description

For my use case I need to compare the first n characters to match a certain "id-like" sequence.

```slint
import { LineEdit } from "std-widgets.slint";

export component Demo inherits Window {
    property <string> id: "Slint";

    preferred-height: 150px;
    preferred-width: 350px;

    background := Rectangle {
        background: {
            // Instead of "==" I need a "starts-with":
            // if input.starts-with("\{root.id}") {
            if input.text == "\{root.id}!" {
                // text starts with "Slint"
                return green;
            }
            purple
        }
    }

    input := LineEdit {
        placeholder-text: "Type 'Slint!' to make background green";
    }
}
```

Is there any support for partial string comparison in Slint? At least I couldn't find anything so far. Personally I think it can be a useful feature to filter certain character sequences in Slint directly before handing it over to the backend. Maybe even a slice based comparison `my-string[1:3] == "123"`. Do you plan to support somthing in this direction?

### Product Impact

_No response_

### PR DESCRIPTION
**Pull request by lzm-build**
Add `starts-with`, `ends-with` to the `string` type.

Fixes: https://github.com/slint-ui/slint/issues/9877

### CODE REVIEW

File: tests/cases/types/string_repeat.slint
Comment (ogoffart) [inline tests/cases/types/string_repeat.slint:1]: Great that there are tests.  That said, I think it would be better to make only one test file. Each test file needs to be compiled separately, better to put them in one file. 
Also add a `out property <bool> test`  tested by the interpreter test.

File: api/cpp/include/private/slint_string.h [hunk 1]
```diff
@@ -160,6 +160,34 @@ struct SharedString
...
+    bool is_ascii() const { return cbindgen_private::slint_shared_string_is_ascii(this); }
+
+    SharedString repeat(const size_t n) const
+    {
+        auto out = SharedString();
+        cbindgen_private::slint_shared_string_repeat(this, n, &out);
+        return out;
+    }
+
+    SharedString replace(const SharedString &from, const SharedString &to) const
```
Comment (ogoffart) [inline api/cpp/include/private/slint_string.h]: Should be `std::string_view` for `to` and `from`

File: api/cpp/include/private/slint_string.h [hunk 2]
```diff
@@ -160,6 +160,34 @@ struct SharedString
         return out;
     }
 
+    bool contains(const SharedString &pat) const
```
Comment (ogoffart) [inline api/cpp/include/private/slint_string.h]: should be `std::string_view`

**Review by ogoffart — COMMENTED** [review]
Thanks for the PR.

What is the motivation behind each of the new call?

For example, what is the use behind is_ascii or repeat?

An use case would help to see if we really want to add these functions.

**Comment by ogoffart** [comment]
The Slint language is meant to be a description of the UI. and not a generic programming language. We should only add feature if they belong on the UI layer. Logic should be done in native code.

Doing input validation probably should be done in native code.
Especially is-ascii would not prevent usernames that contains spaces, or `\n` or various control characters.

File: tests/cases/types/string.slint [hunk 3]
```diff
@@ -14,7 +14,7 @@ export component TestCase  {
     in property<float> increment: 0.1;
     out property <string> converted_value: round(value * 100)/100;
     out property <string> ten_dot_one: 10 + increment;
-    out property <bool> test: e1 && !e2 && converted_value == "98.77" && ten_dot_one == "10.1" && big-number == "800000005";
+    out property <bool> test: e1 && !e2 && converted_value == "98.77" && ten_dot_one == "10.1" && big-number == "800000005" && p1.starts-with("he") && p2.ends-with("🦊");
```
Comment (ogoffart) [inline tests/cases/types/string.slint]: Can we have a bit more tests. Especially  that the function returns false when it should return false.  (eg: covers more than the happy path)
Worth covering is also what happens when passing empty strings and such corner case.

**Review by ogoffart — COMMENTED** [review]
Thanks, i think this looks good now.

I just would like to see a bit more testing.

**Review by ogoffart — APPROVED** [review]
thanks
